custom/inddex/couchdata.py

```diff
@@ -0,0 +1,150 @@
+from memoized import memoized
+
+from corehq.apps.fixtures.models import FixtureDataItem, FixtureDataType
+
+
+class CouchData:
+
+    def __init__(self, domain, tables=None):
+        if not isinstance(tables, tuple):
+            raise TypeError('\'tables\' must be a tuple')
+        self.domain = domain
+        self.tables = {}
+        for table in FixtureDataType.by_domain(domain=domain):
+            if (tables and table.tag in tables) or not tables:
+                self.tables[table.tag] = table.get_id
+
+    @memoized
+    def get_data_from_table(self, table_name=None, fields=None, values=None, as_dict=False):
+        data = []
+        if table_name is None and len(self.tables) == 1:
+            table_name = list(self.tables.keys())[0]
+        elif not table_name:
+            raise AttributeError('\'table_name\' must be specified')
+        records = FixtureDataItem.by_data_type(self.domain, self.tables[table_name])
+        if not fields:
+            return records
+        elif not isinstance(fields, tuple):
+            raise TypeError('\'fields\' must be a tuple')
+        if not isinstance(values, tuple):
+            raise TypeError('\'values\' must be a tuple')
+        values = self._reassign_values({x[0]: x[1] for x in values})
+
+        for record in records:
+            valid_record = self._check_values(record, values)
+            invalid = True if not valid_record else False
+
+            if as_dict and not invalid:
+                data.append(self.record_to_dict(record))
+            elif not invalid:
+                data.append(record)
+
```

Comment:
Records, which don't have values that user is looking for are classified as 'invalid'.